types/src/transaction/transaction.rs

```diff
@@ -264,6 +268,77 @@ fn is_input_and_output_consistent(inputs: &[AssetTransferInput], outputs: &[Asse
     sum.iter().all(|(_, sum)| sum.is_zero())
 }
 
+fn apply_bitmask(
+    bitmask: u128,
+    mut outputs: Vec<AssetTransferOutput>,
+    mut result: Vec<AssetTransferOutput>,
+) -> Vec<AssetTransferOutput> {
+    match bitmask {
+        0 => return result,
+        _ => {
+            if ((bitmask & 0x1) == 1) && !outputs.is_empty() {
+                result.push(outputs.remove(0));
+                apply_bitmask(bitmask >> 1, outputs, result)
+            } else {
+                apply_bitmask(bitmask >> 1, outputs, result)
+            }
+        }
+    }
+}
+
+impl<'a> PartialHashing for &'a Transaction {
```

Comment:
Why do you implement `PartialHashing` for the reference of `Transaction` instead of `Transaction`?

Reply:
Cause when `execute` is called, there is only a borrowed transaction we can use in `fn transfer_asset`.